### DISCUSSION TITLE
unable to add swap to codespace: swapon failed: operation not permitted

### DISCUSSION THREAD

**Discussion by anrondon-microsoft** [community/community#39481] [Codespaces]
### Select Topic Area

Bug

### Body

I have a codespace with 64GB of RAM. Sometimes the ram is full and my development server is killed. One solution that I found was to use a swap file.

I tried to follow the [instructions ](https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-20-04)on how to set a swapfile. but it fails when I try to activate the swap file.

```
sudo swapon /swapfile
swapon: /swapfile: swapon failed: Operation not permitted
```

```
uname -a
Linux codespaces-f6c476 5.4.0-1094-azure #100~18.04.1-Ubuntu SMP Mon Oct 17 11:44:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by mpsenn** [comment]
I'm encountering the same as you, but on the minimum size, 4GB, machine type. Just so you know you aren't alone! Perhaps Github is blocking swapfiles, since a lot of codespaces are free essentially.

**Comment by qc00** [comment]
I needed to add some swap to my codespace as well and stumbled upon this thread.

The problem is `/` is mounted as an overlay file system by docker and the backing FS probably don't support swap files. If you put the file in /tmp then it works just fine.

**Reply by mausch** [reply]
Thanks, this works. Also make sure to make your devcontainer privileged.

BTW note that disks are awfully slow ( https://github.com/community/community/discussions/17616 ) so too much swapping will grind things to a halt.

Would be nice if we could enable zram instead.

**Reply by maggesi** [reply]
@mausch : Can you say what a privileged devcontainer is?

**Comment by memchr** [comment]
Please consider creating a swap file on `/tmp`, which appears to be the mount point of the ext4 filesystem in a virtual SCSI disc, whose throughput is approximately 210 MiB/s
```bash
$ findmnt /tmp
/tmp /dev/sda1[/containerTmp] ext4   rw,relatime
```

Example entry script
```bash
if [[ $(id -u) != 0 ]]; then
	srcpath="$(realpath "$0")"
	exec sudo bash "$srcpath" "$@"
fi

cat > /tmp/mkswap.sh << 'EOF'
dd if=/dev/zero bs=1M of=/tmp/swapfile count=2K
chmod 644 /tmp/swapfile
mkswap /tmp/swapfile
swapon /tmp/swapfile
EOF

bash /tmp/mkswap.sh & disown

exec "$@"
```

**Comment by Zibri** [comment]
@memchr  same.. not permitted.
